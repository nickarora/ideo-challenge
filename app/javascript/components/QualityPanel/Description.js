import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

export const Container = styled.p`
  text-align: left;
  margin-bottom: 3px;
`

const Description = ({ quality }) => (
  <Container>{quality.description}</Container>
)

Description.propTypes = {
  quality: PropTypes.shape({
    description: PropTypes.string.isRequired,
  }).isRequired,
}

export default Description
