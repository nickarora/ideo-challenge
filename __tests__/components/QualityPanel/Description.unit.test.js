import Description from '~/components/QualityPanel/Description'

const longText =
  'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin dignissim in est quis dignissim. Morbi pulvinar risus ac euismod vestibulum. In feugiat felis vel semper.'

describe('Description', () => {
  it('renders a short description without a read more link', () => {
    const wrapper = shallow(
      <Description
        quality={{
          description: 'short description',
        }}
      >
        CONTENT
      </Description>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a long description by truncating and adding a read more link', () => {
    const wrapper = shallow(
      <Description
        quality={{
          description: longText,
        }}
      >
        CONTENT
      </Description>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('tapping readmore expands the full text', () => {
    const wrapper = shallow(
      <Description
        quality={{
          description: longText,
        }}
      >
        CONTENT
      </Description>
    )

    wrapper.find('#readmore-toggle').simulate('click')

    expect(
      wrapper
        .find('#readmore-toggle')
        .render()
        .text()
    ).toEqual('read less')
  })
})
