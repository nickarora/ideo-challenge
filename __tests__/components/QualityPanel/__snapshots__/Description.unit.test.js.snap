// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description renders a long description by truncating and adding a read more link 1`] = `
<styled.section>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin dignissim in est quis dignissim. Morbi pulvinar risus ...
  <styled.div>
    <styled.span
      id="readmore-toggle"
      onClick={[Function]}
    >
      read more
    </styled.span>
  </styled.div>
</styled.section>
`;

exports[`Description renders a short description without a read more link 1`] = `
<styled.section>
  short description
</styled.section>
`;
